packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,268 @@
+import { attr, FASTElement, observable, Updates } from '@microsoft/fast-element';
+import { autoUpdate, computePosition, flip, hide, size } from '@floating-ui/dom';
+import { keyEnter, keyEscape, keySpace, keyTab } from '@microsoft/fast-web-utilities';
+import { MenuList } from '../menu-list/menu-list.js';
+
+/**
+ * The base class used for constructing a fluent-menu custom element
+ * @public
+ */
+export class Menu extends FASTElement {
+  public connectedCallback() {
+    super.connectedCallback();
+    document.addEventListener('click', this.handleDocumentClick);
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    document.removeEventListener('click', this.handleDocumentClick);
+  }
+
+  /**
+   * Defines whether the menu is expanded or not.
+   * @public
+   */
+  @observable
+  @attr({ mode: 'boolean' })
+  public expanded: boolean = false;
+
+  /**
+   * The array of HTMLElements for the menu.
+   * @public
+   */
+  @observable
+  public menu?: HTMLElement[];
+
+  /**
+   * The array of HTMLElements for the trigger.
+   * @public
+   */
+  @observable
+  public trigger?: HTMLElement[];
+
+  /**
+   * Attribute that determines if the menu should open on hover.
+   * @public
+   */
+  @attr({ attribute: 'open-on-hover', mode: 'boolean' })
+  public openOnHover: boolean = false;
+
+  /**
+   * Holds the reference to the MenuList object.
+   * @private
+   */
+  private _menu?: MenuList;
+
+  /**
+   * Holds the reference to the HTMLElement that triggers the menu.
+   * @private
+   */
+  private _trigger?: HTMLElement;
+
+  /**
+   * Container that holds the menu list items.
+   * @public
+   */
+  public menuListContainer?: HTMLElement;
+
+  /**
+   * Holds a reference to a function that is used to cleanup resources.
+   * @private
+   */
+  private cleanup?: () => void;
+
+  /**
+   * Method to toggle the expanded state of the menu on an event.
+   * @param {MouseEvent} e - The event that triggered the expansion of the menu.
+   * @public
+   */
+  public toggleMenu = () => {
+    this.expanded = !this.expanded;
+  };
+
+  /**
+   * Method to close the menu.
+   * @public
+   */
+  public closeMenu() {
+    this.expanded = false;
+  }
+
+  /**
+   * Method to open the menu.
+   * @public
+   */
+  public openMenu() {
+    this.expanded = true;
+  }
+
+  /**
+   * Method to focus on the menu.
+   * @public
+   */
+  public focus(): void {
+    if (this.expanded && this._menu) {
+      const menu = this._menu as MenuList;
+      Updates.enqueue(() => menu.focus());
+    }
+  }
+
+  /**
+   * Method to set the positioning of the menu.
+   * @public
+   */
+  public setPositioning(): void {
```

Comment:
Do we want this publicly available for overriding or should it require a extending the class to do so (protected)? Probably something to discuss more broadly - not a blocker.